scripts/skills/perks/perk_rf_mauler.nut

```diff
@@ -0,0 +1,123 @@
+this.perk_rf_mauler <- this.inherit("scripts/skills/skill", {
+	m = {
+		IsForceEnabled = false,
+		RequireWeapon = true,
+		Chance = 33,
+		InjuryPool = null
+	},
+	function create()
+	{
+		this.m.ID = "perk.rf_mauler";
+		this.m.Name = ::Const.Strings.PerkName.RF_Mauler;
+		this.m.Description = ::Const.Strings.PerkDescription.RF_Mauler;
+		this.m.Icon = "ui/perks/rf_mauler.png";
+		this.m.Type = ::Const.SkillType.Perk;
+		this.m.Order = ::Const.SkillOrder.Perk;
+		this.m.IsActive = false;
+		this.m.IsStacking = false;
+		this.m.IsHidden = false;
+	}
+
+	function onBeforeTargetHit( _skill, _targetEntity, _hitInfo )
+	{
+		this.m.InjuryPool = null;
+
+		if (_targetEntity.getCurrentProperties().IsAffectedByInjuries && ::Math.rand(1, 100) <= this.m.Chance && _targetEntity.getSkills().getSkillByID("effects.bleeding").m.Stacks >= 3)
+			this.m.InjuryPool = _hitInfo.Injuries;
+	}
+
+	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		if (!_targetEntity.isAlive() || !this.isSkillValid(_skill))
+			return;
+
+		if (!_targetEntity.getCurrentProperties().IsImmuneToBleeding)
+			_targetEntity.getSkills().add(::new("scripts/skills/effects/bleeding_effect"));
+
+		if (this.m.InjuryPool == null || (_targetEntity.getFlags().has("undead") && !_targetEntity.getFlags().has("ghoul") && !_targetEntity.getFlags().has("ghost") && !this.getContainer().hasSkill("perk.crippling_strikes")))
+			return;
+
+		// Below is a copy of how vanilla applies injury in actor.nut onDamageReceived function
```

Comment:
The "code below" would be great in a separate function (at the bottom of this script) to make the `onTargetHit` function of this perk much smaller and easier to understand.